cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -767,6 +769,113 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
 }
 }
 
+GANDIVA_EXPORT
+const char* gdv_fn_from_unixtime_int64(int64_t context, gdv_timestamp in,
+                                       int32_t* out_len) {
+  const char* pattern = "yyyy-MM-dd hh:mm:ss";
+  const int length = strlen(pattern);
+  const char* ret =
+      gdv_fn_from_unixtime_int64_utf8(context, in, pattern, length, out_len);
+
+  return ret;
+}
+
+GANDIVA_EXPORT
+const char* gdv_fn_from_unixtime_int64_utf8(int64_t context, gdv_timestamp in,
+                                            const char* pattern, int32_t pattern_len,
+                                            int32_t* out_len) {
+  // Patter dictionary to translate a given pattern like yyyy-MM-dd to
+  // a pattern like %Y-%m-%d that the std::strftime can translate.
+  std::map<std::string, std::string> pattern_dict{
+      {"YYYY", "%Y"},  // converts 'YYYY' to full year, eg. 1970
+      {"YY", "%y"},    // converts 'YY' to final two-digits year, eg. 70
+      {"yyyy", "%Y"},  // converts 'YYYY' to full year, eg. 1970
+      {"yy", "%y"},    // converts 'YY' to final two-digits year, eg. 70
+      {"MM", "%m"},    // converts 'MM' to month digits, eg. 10
+      {"M", "%b"},     // converts 'M' to month abbreviation, eg. Oct
+      {"Mm", "%B"},    // converts 'Mm' to month abbreviation, eg. October
+      {"d", "%e"},  // converts 'd' to day of the month as a decimal number (range [1,31])
+      {"dd",
+       "%d"},  // converts 'dd' to day of the month as a decimal number (range [01,31])
+      {"h",
+       "%I"},  // converts 'h' hour as a decimal number, 12 hour clock (range [01-12])
+      {"hh",
+       "%H"},  // converts 'hh' to hour as a decimal number, 24 hour clock (range [00-23])
+      {"m", "%M"},   // converts 'm' to minute as a decimal number (range [00,59])
+      {"mm", "%M"},  // converts 'mm' minute as a decimal number (range [00,59])
+      {"s", "%S"},   // converts 's' to second as a decimal number (range [00,60])
+      {"ss", "%S"},  // converts 'ss' to second as a decimal number (range [00,60])
+  };
+
+  if (pattern_len <= 0) {
+    gdv_fn_context_set_error_msg(context,
+                                 "Invalid pattern, it must have at least 1 char");
+    *out_len = 0;
+    return "";
+  }
```

Comment:
Is the pattern Literal? If it is Literal you can follow the example of the [TO_DATE](https://github.com/apache/arrow/blob/master/cpp/src/gandiva/to_date_holder.cc) function and move this process to a holder to avoid instantiate and convert this map for each value